src/http/cookie.cr

```diff
@@ -27,6 +27,33 @@ module HTTP
 
     def_equals_and_hash name, value, path, expires, domain, secure, http_only, samesite, extension
 
+    # Parses a `Cookie` from the provided `set-cookie` *header*.
+    #
+    # Raises an `ArgumentError` if the *header* fails to parse.
+    #
+    # ```
+    # require "http/cookie"
+    #
+    # HTTP::Cookie.from_set_cookie_header "id=a3fWa; Expires=Wed, 21 Oct 2015 07:28:00 GMT; Secure; SameSite=Lax" # =>
+    # # <HTTP::Cookie:0x7f684eb29770
+    # # @domain=nil,
+    # # @expires=2015-10-21 07:28:00.0 UTC,
+    # # @extension=nil,
+    # # @http_only=false,
+    # # @name="id",
+    # # @path=nil,
+    # # @samesite=Lax,
+    # # @secure=true,
+    # # @value="a3fWa">
+    # ```
+    def self.from_set_cookie_header(header : String) : self
```

Comment:
What do you think of introducing a nilable version of this? The raised exception doesn't tell you **why** it failed to parse, so raising an exception feels too much compared to just returning `nil` when it failed to parse.

So what I'm proposing is either:
1. Change this to return `nil` instead of raising an error
2. Let it return an exception, but let the exception say why it failed to parse (and optionally include a nilable version)

Reply:
That said, `String#to_i` also doesn't explain why it failed to parse, so feel free to ignore me.